src/chrome/local-runtime.ts

```diff
@@ -208,6 +211,34 @@ export default class LocalRuntime {
     }
   }
 
+  // Returns the S3 url or local file path
+  async returnPDF(): Promise<string> {
+    const data = await pdf(this.client)
+
+    // check if S3 configured
+    if (process.env['CHROMELESS_S3_BUCKET_NAME'] && process.env['CHROMELESS_S3_BUCKET_URL']) {
+      const s3Path = `${cuid()}.pdf`
+      const s3 = new AWS.S3()
+      await s3.putObject({
+        Bucket: process.env['CHROMELESS_S3_BUCKET_NAME'],
+        Key: s3Path,
+        ContentType: 'application/pdf',
+        ACL: 'public-read',
+        Body: new Buffer(data, 'base64'),
+      }).promise()
+
+      return `https://${process.env['CHROMELESS_S3_BUCKET_URL']}/${s3Path}`
+    }
+
+    // write to `/tmp` instead
+    else {
+      const filePath = `/tmp/${cuid()}.pdf`
```

Comment:
With node 4 and above [os.tmpdir()](https://nodejs.org/dist/latest-v4.x/docs/api/os.html#os_os_tmpdir) should do the trick.